### PR TITLE
fix(mu4e): cooperative lock file watching

### PR DESCRIPTION
**Pull request by tecosaur**
Some time ago I noticed the cooperative lock file management wasn't working as I remember. I forget what exactly I was thinking, but basically I've poked at the code until it seems to work better.

If some other people could test this and report back, that would be much appreciated.

-----
- [x] I searched the issue tracker and this hasn't been PRed before.
- [x] My changes are not on [the do-not-PR list](https://doomemacs.org/d/do-not-pr) for this project.
- [x] My commits conform to [the git conventions](https://doomemacs.org/d/git-conventions).
- [ ] My changes are visual; I've included before and after screenshots.
- [ ] I am blindly checking these off.
- [ ] Any relevant issues or PRs have been linked to.
- [ ] This a draft PR; I need more time to finish it.

### CODE REVIEW

File: modules/email/mu4e/autoload/mu-lock.el [hunk 1]
```diff
@@ -53,12 +53,12 @@ Else, write to this process' PID to the lock file"
       (user-error "Unfortunately another Emacs is already doing stuff with Mu4e, and you can only have one at a time")
     (write-region (number-to-string (emacs-pid)) nil +mu4e-lock-file)
     (delete-file +mu4e-lock-request-file)
-    (call-process "touch" nil nil nil +mu4e-lock-request-file)
```
Comment (seanfarley) [inline modules/email/mu4e/autoload/mu-lock.el:56]: Won't this have the same problem as before on BSD kqueue systems? `file-notify-add-watch` is implementation dependent and doesn't always create the file.

Comment (seanfarley) [reply modules/email/mu4e/autoload/mu-lock.el:56]: Just confirmed:

```
error in process filter: file-notify-add-watch: File does not exist: No such file or directory, /var/folders/bj/cvgkd64s42v_dt0z1bpy397c0000gn/T/mu4e_lock_request
error in process filter: File does not exist: No such file or directory, /var/folders/bj/cvgkd64s42v_dt0z1bpy397c0000gn/T/mu4e_lock_request
```

Comment (seanfarley) [reply modules/email/mu4e/autoload/mu-lock.el:56]: Maybe we can add a comment about `kqueue` differences so as to not remove the call to `touch`?

Comment (tecosaur) [reply modules/email/mu4e/autoload/mu-lock.el:56]: Hmm, how do you find the behaviour if that line is kept? (i.e. does it work)

Comment (seanfarley) [reply modules/email/mu4e/autoload/mu-lock.el:56]: It works as expected if I add back the `touch` call.

Comment (seanfarley) [reply modules/email/mu4e/autoload/mu-lock.el:56]: Alternatively, we could patch the emacs functions to create the file to make all the implementations line up. Maybe even submit that upstream.

**Comment by seanfarley** [comment]
image

**Comment by tecosaur** [comment]
Sean ... where did you get all of these images from?

**Review by hlissner — APPROVED** [review]

**Comment by hlissner** [comment]
Thanks again!

**Pull request merged**